src/components/Badge/Badge.st.css

```diff
@@ -0,0 +1,316 @@
+:import {
+  -st-from: "wix-ui-core/dist/src/components/StylableBadge/BadgeStyle.st.css";
+  -st-default: Badge;
+}
+
+:import {
+  -st-from: "wix-ui-core/dist/src/components/StylableBadge/Badge.st.css";
+  -st-default: BaseBadge;
+}
+
+:import {
+  -st-from: "../../palette.st.css";
+  -st-named: secondaryText, primaryLightText, mainHover, danger, success, placeholderText, warning, urgent, mainMuted, successMutedHover, backgroundSecondary, dangerMutedHover;
+}
+
+.root {
+  -st-extends: BaseBadge;
+  -st-states: skin(string), type(string);
+}
+
+.root >:not(:last-child) {
+  margin-right: 8px;
+}
+
+.prefix, .suffix {
+  width: 12px;
+  height: 12px;
+}
+
+/* children color when type = solid */
+.root:type(solid):skin(default) .text,
+.root:type(solid):skin(standard) .text,
+.root:type(solid):skin(danger) .text,
+.root:type(solid):skin(success) .text,
+.root:type(solid):skin(grey) .text,
+.root:type(solid):skin(warning) .text,
+.root:type(solid):skin(urgent) .text {
+  color: value(primaryLightText);
+}
+
+.root:type(solid):skin(default) .prefix svg,
+.root:type(solid):skin(standard) .prefix svg,
+.root:type(solid):skin(danger) .prefix svg,
+.root:type(solid):skin(success) .prefix svg,
+.root:type(solid):skin(grey) .prefix svg,
+.root:type(solid):skin(warning) .prefix svg,
+.root:type(solid):skin(urgent) .prefix svg,
+
+.root:type(solid):skin(default) .suffix svg,
+.root:type(solid):skin(standard) .suffix svg,
+.root:type(solid):skin(danger) .suffix svg,
+.root:type(solid):skin(success) .suffix svg,
+.root:type(solid):skin(grey) .suffix svg,
+.root:type(solid):skin(warning) .suffix svg,
+.root:type(solid):skin(urgent) .suffix svg {
+  fill: value(primaryLightText);
+}
+
+.root:type(solid):skin(neutralStandard) .text,
+.root:type(solid):skin(neutralSuccess) .text,
+.root:type(solid):skin(neutralDanger) .text {
+  color: value(secondaryText);
+}
+
+.root:type(solid):skin(neutralStandard) .prefix svg,
+.root:type(solid):skin(neutralSuccess) .prefix svg,
+.root:type(solid):skin(neutralDanger) .prefix svg,
+
+.root:type(solid):skin(neutralStandard) .suffix svg,
+.root:type(solid):skin(neutralSuccess) .suffix svg,
+.root:type(solid):skin(neutralDanger) .suffix svg {
+  fill: value(secondaryText);
+}
+
+/* children color when type != solid */
+.root:skin(default) .text {
+  color: value(secondaryText);
+}
+
+.root:skin(default) .prefix svg,
```

Comment:
@alex5il is this not a good selector, because it can be not only svg?

The usage in the tsx file is:
```js
<span className={style.prefix}><SomeIcon/></span>
```

Reply:
you can pass in `className` to icons components generated with `svg2react-icon`

Reply:
And what if it is not generated with react-svg? is it ok to assume that in the bo it will always be an svg?